src/slots/Slot.js

```diff
@@ -183,10 +206,15 @@ const getIncomingDataStreams = (slot, fctId) => (
 )
 
 const _assertSlotDataTypesCompatible = (slotA, slotB) => {
+  console.log({slotA: slotA.dataType,slotB: slotB.dataType});
+
+  const isCompatible = slotA.dataType === slotB.dataType
```

Comment:
i found the other version made my head hurt so this is more explicit

Reply:
i think the logic is right

same datatype is good 
if either or both are any then its good 
if one is a number the other one needs to be ANY_NUMBER (unless its the same type which is covered by case 1)